app/models/lookup_value/puppet_lookup_value.rb

```diff
@@ -0,0 +1,15 @@
+class PuppetLookupValue < LookupValue
+  attr_accessible :use_puppet_default
+  validate :puppet_lookup_key?
+  validate :value_present?
+
+  private
+
+  def puppet_lookup_key?
```

Comment:
Simply call this `puppet?` for consistency with the lookup key?  You can `delegate` it then too.